manifests/job.pp

```diff
@@ -25,12 +25,13 @@
 #     choose 'absent' to ensure the job is removed
 #
 define jenkins::job(
-  $config,
+  $config   = undef,
```

Comment:
This changes the API of this defined type and makes `$config` optional when it's supposed to be required.